nupic/algorithms/anomaly.py

```diff
@@ -133,9 +134,8 @@ def compute(self, activeColumns, predictedColumns,
     if self._mode == Anomaly.MODE_PURE:
       score = anomalyScore
     elif self._mode == Anomaly.MODE_LIKELIHOOD:
-      if inputValue is None:
-        raise ValueError("Selected anomaly mode 'Anomaly.MODE_LIKELIHOOD' "
-                 "requires 'inputValue' as parameter to compute() method. ")
+      assert inputValue is not None, "Selected anomaly mode \
```

Comment:
on the contrary to the other cases, I think here assert should stay. compute() is executed each iteration, and the check is "just for extra safety", it's fine to optimize it out in production.